classes/core/PKPEventServiceProvider.inc.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+use Illuminate\Support\Facades\Event;
+use Illuminate\Events\EventServiceProvider;
+
+class PKPEventServiceProvider extends EventServiceProvider {
+
+	/**
+	 * @var array $listen $event => $listeners[]
+	 * @brief Registering events & listeners, see Illuminate\Events\EventServiceProvider
+	 */
+	protected $listen = [];
+
+	/**
+	 * @var array
+	 * @brief to load subscriber classes, currently empty
+	 */
+	protected $subscribe = [];
+
+	/**
+	 * @return void;
+	 * @brief boot the service after registration
+	 */
+	public function register() {
+		parent::register();
+	}
+
+	/**
+	 * Get the discovered events and listeners for the application.
+	 * TODO handle cached events
+	 * @return array
+	 */
+	public function getEvents()
+	{
+		return $this->listens();
+
+	}
+
+	/**
+	 * Get the events and handlers.
+	 *
+	 * @return array
+	 */
+	public function listens()
+	{
+		return $this->listen;
+	}
+
+	/**
+	 * @brief Boot events
+	 */
+	public function boot() {
+
+		$events = $this->getEvents();
+
+		foreach ($events as $event => $listeners) {
+			foreach (array_unique($listeners) as $listener) {
+				Event::listen($event, $listener);
+			}
+		}
+
+		foreach ($this->subscribe as $subscriber) {
+			Event::subscribe($subscriber);
+		}
+	}
+}
```

Comment:
Does this file do anything that Laravel's `EventServiceProvider` doesn't do? Can we leave it out for now?

Reply:
Yes, all those methods are overloaded deliberately. Laravel registers EventServiceProvide 2 times, first time bindings and the second time on the application level to pass event data and settings. I'm doing it all at once here and need to bypass some of Laravel's application-level logic, which we don't have.

Reply:
To be more precise, those methods and properties aren't defined on the container layer service provider. And needed to be implemented on the application level.

Reply:
Got it, thanks!